scripts/main/album.js

```diff
@@ -27,7 +35,7 @@ album.isModelID = function (id) {
  * @returns {?string}
  */
 album.getParentID = function () {
-	if (album.json == null || album.isSmartID(album.json.id) === true || !album.json.parent_id) {
+	if (album.json === null || album.isSmartID(album.json.id) || album.isSearchID(album.json.id) || !album.json.parent_id) {
```

Comment:
```suggestion
	if (album.json === null || album.isSmartID(album.json.id) || /* album.isSearchID(album.json.id) || */ !album.json.parent_id) {
```

While this will eventually be a right change, for now we need to disable it because as things stand, this makes it impossible to click on photos in the search results. I'll deal with it as part of my search-fixing PR.

Reply:
Even though I do not understand why this code change makes a difference, I do accept that it does.

I expected the result to be the same. If `album.json.id === 'search'` holds, then `album.json` is an instance of `SearchAlbum`. This in turn implies that `album.json.parent_id` equals `null`, because `SearchAlbum` has no parent. So in this case, it should actually not matter whether we return `null` early (due to the `if`-condition)  or if we return `null` later (due to `album.json.parent_id` being `null`).

As experience suggests, this assumptions is not true. So "something" somehow somewhere overwrites `album.json.paren_id` with some non-null value. This frontend code is sick. There is always some code which cannibalize objects in some non-obvious way to implicitly achieve some effect later. That alone is bad, but the lack of any comments that would point that out is horrible.

Reply:
After I had accepted your suggestion, I realized that it did not solve the problem. The modification must be applied to `album.getID` not `album.getParentID`. See https://github.com/LycheeOrg/Lychee-front/pull/282/commits/62e2ac328c5ac6f003e3756fdac19690c112c9f6. This make also much more sense, as `album.getID` already contains some code which handles the `SearchAlbum` specifically.

Reply:
Yeah, sorry about that. That's the problem with these suggestions: you have to reenter them on the website instead of doing a git push, which increases a chance of error...